src/test/java/org/mskcc/cbio/oncokb/web/rest/UsageAnalysisControllerIT.java

```diff
@@ -0,0 +1,931 @@
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.testcontainers.shaded.org.apache.commons.io.IOUtils;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.http.AbortableInputStream;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+
+/**
+ * Unit tests for the {@link UsageAnalysisController } REST controller.
+ */
+public class UsageAnalysisControllerIT {
+  private final Gson gson = new Gson();
+
+  @Mock
+  private S3Service s3Service;
+
+  @Mock
+  private UserService userService;
+
+  @Mock
+  private UserMapper userMapper;
+
+  @InjectMocks
+  private UsageAnalysisController usageAnalysisController;
+
+  private MockMvc restMockMvc;
+
+  private final MockS3Data data = new MockS3Data(1);
+
+  public UsageAnalysisControllerIT() throws Exception {}
+
+  private void mockS3ObjectResponse(MockS3Data data) throws Exception {
+    Mockito
+      .when(userMapper.userToUserDTO(any()))
+      .thenAnswer(
+        i -> {
+          User user = (User) i.getArguments()[0];
+          Optional<UserDTO> userDto = data.userDtos
+            .stream()
+            .filter(x -> Objects.equals(x.getId(), user.getId()))
+            .findFirst();
+          return userDto.orElseThrow(Exception::new);
+        }
+      );
+
+    Mockito
+      .when(userService.getUserWithAuthoritiesByEmailIgnoreCase(any()))
+      .thenAnswer(
+        i -> {
+          String email = (String) i.getArguments()[0];
+          return data.users
+            .stream()
+            .filter(x -> x.getEmail().equalsIgnoreCase(email))
+            .findFirst();
+        }
+      );
+
+    Mockito
+      .when(userService.getUserById(any()))
+      .thenAnswer(
+        i -> {
+          long id = (long) i.getArguments()[0];
+          return data.users.stream().filter(x -> x.getId() == id).findFirst();
+        }
+      );
+
+    Mockito
+      .when(s3Service.getObject(eq(Constants.ONCOKB_S3_BUCKET), any()))
+      .thenAnswer(
+        i -> {
+          String filePath = (String) i.getArguments()[1];
+          JsonElement element = data.files.get(filePath);
+          String fileContents = gson.toJson(element);
+          InputStream objectStream = IOUtils.toInputStream(
+            fileContents,
+            "UTF-8"
+          );
+          ResponseInputStream<GetObjectResponse> stream = new ResponseInputStream<>(
+            GetObjectResponse.builder().build(),
+            AbortableInputStream.create(objectStream)
+          );
+          return Optional.of(stream);
+        }
+      );
+  }
+
+  @BeforeEach
+  public void setup() throws Exception {
+    usageAnalysisController = new UsageAnalysisController();
+    MockitoAnnotations.initMocks(this);
+    this.mockS3ObjectResponse(data);
+    this.restMockMvc =
+      MockMvcBuilders.standaloneSetup(usageAnalysisController).build();
+  }
+
+  @Test
+  public void shouldGetUsageSummaryResources() throws Exception {
```

Comment:
Am I correct to say these tests are to test whether the endpoints will return previously generated files properly?

Reply:
Yeah they're to test what the current behavior of the endpoints are.